src/library_fs.js

```diff
@@ -1063,10 +1063,13 @@ mergeInto(LibraryManager.library, {
       var stream = FS.createStream({
         node: node,
         path: FS.getPath(node),  // we want the absolute path to the node
+        id: node.id,
         flags: flags,
+        mode: node.mode,
         seekable: true,
         position: 0,
         stream_ops: node.stream_ops,
+        node_ops: node.node_ops,
         // used by the file family libc calls (fopen, fwrite, ferror, etc.)
```

Comment:
To check I understand: this is copying fields off of the `node`, because there isn't a node in the directory case, or something like that?

What puzzles me is that I don't see new code doing (for example) `.mode` in this diff. Where is `stream.mode` read?

Reply:
The `node_ops` and `mode` properties are necessary for the `FS.lookupNode(stream, name);` call in `SYS_getdents64`.

`node_ops` path:

- https://github.com/AlexanderOMara/emscripten/blob/bug-opendir-readdir-master/src/library_syscall.js#L1057
- https://github.com/AlexanderOMara/emscripten/blob/bug-opendir-readdir-master/src/library_fs.js#L182
- https://github.com/AlexanderOMara/emscripten/blob/bug-opendir-readdir-master/src/library_fs.js#L335

`mode` path:

- https://github.com/AlexanderOMara/emscripten/blob/bug-opendir-readdir-master/src/library_syscall.js#L1057
- https://github.com/AlexanderOMara/emscripten/blob/bug-opendir-readdir-master/src/library_fs.js#L182
- https://github.com/AlexanderOMara/emscripten/blob/bug-opendir-readdir-master/src/library_fs.js#L323